sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1646,6 +1676,11 @@ public WriteResult expand(PCollection<T> input) {
             "The supplied getTableFunction object can directly set TimePartitioning."
                 + " There is no need to call BigQueryIO.Write.withTimePartitioning.");
       }
+      if (getJsonClustering() != null) {
+        checkArgument(
+            getJsonTimePartitioning() != null,
```

Comment:
So that combinations (getJsonClustering and getTimePartitioning) and (getClustering and getJsonTimePartitioning) are not accepted ?

Reply:
It should be covered now if I'm not mistaken